webui/cypress/support/step_definitions/logins.steps.ts

```diff
@@ -14,13 +14,17 @@
  * limitations under the License.
  */
 
-import { Given } from 'cypress-cucumber-preprocessor/steps';
+import { Given, Then } from 'cypress-cucumber-preprocessor/steps';
 
 import { LoginPage } from '../../pages/login.po';
 
 const page = new LoginPage();
 
-Given('I logged in', async () => {
+Given('I log(ged) in', async () => {
   await page.login();
 });
 
+Then('my avatar image should be set', async () => {
+  page.getAvatar().should('exist');
+});
```

Comment:
@zsxking do you have advice on how to write a better assertion here? The avatar could either be an `<img>` or a `<mat-icon>` and I couldn't find documentation on how to do an `or` assertion.

Reply:
As far as the e2e test, I think that's really as good as it could be. The next step for improving will be making sure if the user used in e2e actually have avatar photo or not.

But honestly, I think this will be better tested with unit test on app.components. You will be able to control the state, as well as stubbing out the `onAuthStateChanged` (I think) and assert the rendering accordingly.